ssl/record/ssl3_record.c

```diff
@@ -663,6 +663,15 @@ int ssl3_get_record(SSL *s)
             goto f_err;
         }
 
+        /* If client see a packet overflowing his requested MFL setting */
```

Comment:
But the negotiated MFL, once negotiated, applies to *both* peers:

    Once a maximum fragment length other than 2^14 has been successfully
    negotiated, the client and server MUST immediately begin fragmenting
    messages (including handshake messages) to ensure that no fragment
    larger than the negotiated length is sent.